src/test/java/com/javatemplate/persistent/book/BookStoreTest.java

```diff
@@ -27,8 +36,95 @@ void shouldFindAll_OK() {
         when(bookRepository.findAll())
                 .thenReturn(expected);
 
-        assertEquals(expected.size(), bookStore.findAll().size());
+        final var actual = bookStore.findAll();
+
+        assertEquals(expected.size(), actual.size());
 
         verify(bookRepository).findAll();
     }
-}
\ No newline at end of file
+
+    @Test
+    void shouldSave_OK() {
+        final var book = buildBookEntity();
+
+        when(bookRepository.save(any())).thenReturn(book);
+
+        final var expected = bookStore.save(toBook(book));
+
+        assertEquals(book.getId(), expected.getId());
+        assertEquals(book.getName(), expected.getName());
+        assertEquals(book.getAuthor(), expected.getAuthor());
+        assertEquals(book.getUpdatedAt(), expected.getUpdatedAt());
+        assertEquals(book.getCreatedAt(), expected.getCreatedAt());
+        assertEquals(book.getDescription(), expected.getDescription());
+        assertEquals(book.getImage(), expected.getImage());
+    }
+
+    @Test
+    void shouldDeleteById_OK() {
+        final var book = buildBookEntity();
+
+        bookStore.deleteById(book.getId());
+
+        verify(bookRepository).deleteById(book.getId());
+    }
+
+    @Test
+    void shouldFindById_OK() {
+        final var book = buildBookEntity();
+        final var bookOpt = Optional.of(book);
+
+        when(bookRepository.findById(book.getId())).thenReturn(bookOpt);
+
+        final var actual = bookStore.findById(book.getId()).get();
+        final var expected = bookOpt.get();
+
+
+        assertEquals(actual.getId(), expected.getId());
+        assertEquals(actual.getName(), expected.getName());
+        assertEquals(actual.getAuthor(), expected.getAuthor());
+        assertEquals(actual.getUpdatedAt(), expected.getUpdatedAt());
+        assertEquals(actual.getCreatedAt(), expected.getCreatedAt());
+        assertEquals(actual.getDescription(), expected.getDescription());
+        assertEquals(actual.getImage(), expected.getImage());
+
+        verify(bookRepository).findById(book.getId());
+    }
+
+    @Test
+    void shouldFindById_Empty() {
+        final var uuid = randomUUID();
+
+        when(bookRepository.findById(uuid)).thenReturn(Optional.empty());
+
+        assertFalse(bookStore.findById(uuid).isPresent());
+        verify(bookRepository).findById(uuid);
+    }
+
+    @Test
+    void shouldBooksFindByName_OK() {
+        final var book = buildBookEntity();
+        final var expected = buildBookEntities();
+
+        when(bookRepository.findBooksByName(anyString())).thenReturn(expected);
+
+        final var actual = bookStore.findBooksByName(book.getName());
+
+        assertEquals(actual.size(), expected.size());
+
+        verify(bookRepository).findBooksByName(book.getName());
+    }
+
+    @Test
+    void shouldBooksFindByName_Empty() {
+        final var bookName = randomAlphabetic(3, 10);
+
+        when(bookRepository.findBooksByName(bookName)).thenReturn(Collections.emptyList());
```

Comment:
Import static nếu có thể
```suggestion
        when(bookRepository.findBooksByName(bookName)).thenReturn(emptyList());
```

Reply:
Fixed